### PR TITLE
able to use the pulp cert and key to auth

### PR DESCRIPTION
**Pull request by LinglingXiang**
able to use the pulp cert and key to auth

### CODE REVIEW

**Comment by LinglingXiang** [comment]
@rbikar ,please help review, thanks

File: tests/integration/test_association.py [hunk 1]
```diff
@@ -41,16 +42,24 @@ def load_ubiconfig(filename, version):
 
 @pytest.fixture(name="pulp_client")
 def make_pulp_client():
-    with Client(
-        url="https://" + PULP_HOSTNAME + "/",
-        auth=(PULP_USER, PULP_PWD),
-        verify=PULP_SECURE,
-    ) as client:
-        yield client
+    if PULP_CERT_PATH is not None:
```
Comment (rbikar) [inline tests/integration/test_association.py:45]: Can you de-duplicate the code please?
Check this out for one of possible variant:
https://github.com/release-engineering/ubi-manifest/blob/62f3e4d55ee61f389b69ad9efb6b35c726092bf2/ubi_manifest/worker/tasks/depsolver/utils.py#L21

**Review by rbikar — CHANGES_REQUESTED** [review]

File: tests/integration/test_association.py (hunk 1 above)
Comment (LinglingXiang) [reply tests/integration/test_association.py]: changed.

File: tests/integration/test_association.py [hunk 2]
```diff
@@ -73,7 +82,12 @@ def run_ubipop_tool(content_set, workers=10, dry_run=False):
 
 
 def get_repos_from_cs(cs, skip_dot_version=False):
-    p = Pulp(PULP_HOSTNAME, auth=(PULP_USER, PULP_PWD), verify=PULP_SECURE)
+    if PULP_CERT_PATH is None:
```
[the remark this answers is not in the document]
Comment (LinglingXiang) [reply tests/integration/test_association.py]: hi Robert, i didn't change it as in future, i want to replace to use the same pubtools.pulplib client as above.

File: tests/integration/test_association.py [hunk 3]
```diff
@@ -41,16 +42,24 @@ def load_ubiconfig(filename, version):
...
+            url="https://" + PULP_HOSTNAME + "/",
+            auth=(PULP_USER, PULP_PWD),
+            verify=PULP_SECURE,
+        ) as client:
+            yield client
 
 
 def run_ubipop_tool(content_set, workers=10, dry_run=False):
-    if PULP_CERT_PATH is None:
+    if PULP_CERT_PATH is not None:
```
[the remark this answers is not in the document]
Comment (LinglingXiang) [reply tests/integration/test_association.py]: changed.

File: tests/integration/test_association.py [hunk 4]
```diff
@@ -41,16 +42,24 @@ def load_ubiconfig(filename, version):
...
         auth = (PULP_USER, PULP_PWD)
     else:
         auth = (PULP_CERT_PATH,)
```
[the remark this answers is not in the document]
Comment (LinglingXiang) [reply tests/integration/test_association.py]: changed

**Comment by rbikar** [comment]
run tests

**Comment by LinglingXiang** [comment]
hi Robert, the #175 failed as mongo service on pulp server, so we can see the db restore failed on the job .after fix the service of mongo, the #176 is pass

**Review by rbikar — APPROVED** [review]

**Pull request merged**